CMakeLists.txt

```diff
@@ -66,11 +66,20 @@ endif( )
 # hipSPARSE project
 project(hipsparse LANGUAGES CXX ${fortran_language})
 
-# Build flags
-set(CMAKE_CXX_STANDARD 14)
+# Set CXX flags
+if (NOT DEFINED CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+endif()
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+# Set CXX standard
+if (CMAKE_CXX_STANDARD EQUAL 14)
+  message(WARNING "C++14 will be deprecated in the next major release")
+elseif(NOT CMAKE_CXX_STANDARD EQUAL 17)
```

Comment:
I think your intent is something more like this:
message( DEPRECATION "Builds using the C++14 standard will no longer be supported in the next major release" )
You are deprecating support for C++14 with this PR, not in the future

Reply:
Good point. Fixed.